### ISSUE TITLE
OpenGL version detection is too fragile

### ISSUE THREAD

**Issue by jwhear** [mdparker/Derelict3#116]
The current OpenGL version detection routine in gl3.d is not robust enough and fails with some drivers which report version numbers with a patch level.  E.g. with linux 64-bit NVIDIA 304.88, `glGetString(GL_VERSION)` returns the following: "3.0.0 NVIDIA 304.88" which causes `findMaxAvailable` to return HighestSupported, which in turn leads to "failed to bind symbol ..." errors.

**Comment by mdparker** [comment]
Did https://github.com/aldacron/Derelict3/pull/112 not fix that?

**Comment by jwhear** [comment]
No, that fix truncates the "NVIDIA 304.88" bit, the problem is that the version switch block (https://github.com/aldacron/Derelict3/blob/master/import/derelict/opengl3/gl3.d#L515) handles major-minor versions (e.g. "3.0"), but `verstr` is "3.0.0".
I've got a patch that works for me, but it's not particularly elegant, so I'm hesitant to add a pull request: https://github.com/jwhear/Derelict3/commit/eb8930d2a3c780426b9558a13e66247e0f1003f2

**Comment by mdparker** [comment]
OK, I see what you mean. I need to pay more attention when I merge these pull requests. I think using std.string.indexOf will be sufficient.

**Comment by mdparker** [comment]
Actually, there's no reason even to convert the version string to a D string. I'll get a permanent fix for this committed after I get to my box at home in a few hours.

**Comment by mdparker** [comment]
I've committed what I hope is a fix for this. It's entirely untested at the moment because I've got some sort of problem with my build environment. It's late here and I'm off to bed, but I wanted to get this pushed because this issue is a big one (should have gotten to it sooner). If it doesn't work, I'm sure I'll hear about it. But I believe it should. I'll keep this open until I can figure out what's up with my environment and can test it, or hear until I hear from someone that it works.
